controller/RestDelivery.php

```diff
@@ -137,79 +125,99 @@ public function generateDeferred()
     }
 
     /**
-     * Update delivery by id
+     * Update delivery by parameters
      */
     public function update()
     {
         try {
             if ($this->getRequestMethod() !== \Request::HTTP_POST) {
                 throw new \common_exception_NotImplemented('Only post method is accepted to updating delivery');
             }
-            if (!$this->hasRequestParameter(self::REST_DELIVERY_ID)) {
-                throw new \common_exception_MissingParameter(self::REST_DELIVERY_ID, $this->getRequestURI());
+
+            $searchParams = $_GET;
+            $where = [];
+            if ($searchParams) {
+                foreach ($searchParams as $key => $value) {
+                    $rdfKey = \tao_helpers_Uri::decode($key);
+                    $value = \tao_helpers_Uri::decode($value);
+                    $where[$rdfKey] = $value;
+                }
             }
-            $delivery = new \core_kernel_classes_Resource($this->getRequestParameter(self::REST_DELIVERY_ID));
-
-            $queueService = $this->getServiceManager()->get(Queue::SERVICE_ID);
-            if ($taskResource = $queueService->getTaskResource($delivery)) {
-                /** @var Task $task */
-                $task = $queueService->getTask($taskResource->getUri());
-                if ($task
-                    && ($task->getStatus() == Task::STATUS_CREATED
-                        || $task->getStatus() == Task::STATUS_RUNNING
-                        || $task->getStatus() == Task::STATUS_STARTED)
-                ) {
-                    $report = Report::createInfo(__('Compilation of delivery is in progress.'));
-                    $this->returnReport($report);
-                    return;
-                } else if ($task && $task->getStatus() == Task::STATUS_FINISHED) {
-                    /** @var \common_report_Report $report */
-                    $report = $queueService->getReportByLinkedResource($delivery);
-                    if ($report->getType() == Report::TYPE_ERROR) {
-                        $this->returnReport($report);
-                        return;
+            $deliveryModelClass = $this->getDeliveryRootClass();
+            $deliveries = $deliveryModelClass->searchInstances($where, ['like' => false, 'recursive' => true]);
+
+            $propertyValues = $_POST;
+            $response = [];
+
+            /** @var \core_kernel_classes_Resource $delivery */
+            foreach ($deliveries as $key => $delivery) {
+                $queueService = $this->getServiceManager()->get(Queue::SERVICE_ID);
```

Comment:
I think we don't have to check task statuses here. If delivery resource exists it should be enough to update it's properties.